100-level-AsyncAPI-Intro/step2.md

```diff
@@ -0,0 +1,244 @@
+
+2. **Lay the Cosmic Foundations**: Enter the basic AsyncAPI structure:
+   ```yaml
+   asyncapi: 3.0.0
+   info:
+     title: Space Communication API
+     version: '1.0.0'
+     description: API for communication between Chan's spaceship and Capuccinova
+
+   servers:
+     production:
+       host: space.mosquitto.org
+       protocol: mqtt
+       description: Production server for space communication
+
+   channels:
+     roomidchat:
+       address: chat/{roomId}
+       messages:
+         chatMessage:
+           name: ChatMessage
+           payload:
+             type: object
+             properties:
+               userId:
+                 type: string
+                 description: The ID of the user who sent the message
+               message:
+                 type: string
+                 description: The chat message
+               spacename:
+                 type: string
+                 description: The unique spacename identifier for the message
+
+   operations:
+     sendChatMessage:
+       action: 'send'
+       summary: Send a chat message to a specific room
+       channel:
+         $ref: '#/channels/roomidchat'
+   ```
+3. **Seal the Galactic Document**: Save (`Ctrl + O`, `Enter`) and exit Nano (`Ctrl + X`).
+
+4. **Inform Chan**: Let him know that the foundation for his AsyncAPI document is set.
+
+#### Chapter 2: Expanding the Network to Brownieterra
+
+**Your Mission: Adding a New Channel and Defining Operations for Brownieterra**
+
+After successfully laying the groundwork for communication with Capuccinova, Chan is now eager to extend his AsyncAPI document to include a channel for Brownieterra.
+
+1. **Reopen the AsyncAPI Document**: Use the command `nano chan-to-capuccinova.yaml` to edit Chan's document.
+
+2. **Create a New Communication Channel**: Add a channel for Brownieterra. This includes specifying the channel name, its address, and parameters:
+   ```yaml
+   brownieterra-chat:
+     description: Channel for sending messages to Brownieterra
+     address: brownieterra-chat/{spacename}
+     parameters:
+       spacename:
+         description: The unique spacename identifier for Brownieterra communication
+         schema:
+           type: string
+   ```
+
+3. **Define Operations for the New Channel**: Under the `operations` section, specify how messages will be sent over the new Brownieterra channel. For example:
+   ```yaml
+   operations:
+     sendToBrownieterra:
+       action: send
+       summary: Send a message to Brownieterra
+       channel:
+         $ref: '#/channels/brownieterra-chat'
+   ```
+
+4. **Save and Exit**: After adding the new channel and defining its operations, save your changes (`Ctrl + O`, `Enter`) and exit Nano (`Ctrl + X`).
+
+5. **Inform Chan**: Let Chan know that the new channel for Brownieterra has been successfully added and configured in his AsyncAPI document.
```

Comment:
```suggestion
1. Run the command `nano chan-to-capuccinova.yaml` to edit Chan's document.

2. Add a channel for Brownieterra, specifying the channel name, its address, and parameters:
   ```yaml
   brownieterra-chat:
     description: Channel for sending messages to Brownieterra
     address: brownieterra-chat/{spacename}
     parameters:
       spacename:
         description: The unique spacename identifier for Brownieterra communication
         schema:
           type: string
   ```

3. Under the `operations` section, specify how messages will be sent over the new Brownieterra channel.
   ```yaml
   operations:
     sendToBrownieterra:
       action: send
       summary: Send a message to Brownieterra
       channel:
         $ref: '#/channels/brownieterra-chat'
   ```

4. Save your changes (`Ctrl + O`, `Enter`) and exit Nano (`Ctrl + X`).
```

I'm unsure if we actually need step 4, saving changes/closing nano. Does not seem needed if this is in Killercoda.